### PR TITLE
Added tests for SOAP Pass Through APIs

### PR DESCRIPTION
**Pull request by binodmx**
Test creating and publish SOAP Pass Through API using WSDL definition URL
Test downloading WSDL definition from Publisher/Devportal
Test invoking SOAP endpoint

### CODE REVIEW

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/soap/SOAPPassThroughTestCase.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+                new Object[]{TestUserMode.TENANT_ADMIN},
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init(userMode);
+
+        // Load request/response body
+        wsdlDefinition = IOUtils.toString(getClass().getClassLoader().getResourceAsStream("soap" + File.separator +
```
[the remark this answers is not in the document]
Comment (binodmx) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/soap/SOAPPassThroughTestCase.java]: resolved

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/soap/SOAPPassThroughTestCase.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    @Factory(dataProvider = "userModeDataProvider")
+    public SOAPPassThroughTestCase(TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+        return new Object[][]{
+                new Object[]{TestUserMode.SUPER_TENANT_ADMIN},
+                new Object[]{TestUserMode.TENANT_ADMIN},
```
Comment (prasa7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/soap/SOAPPassThroughTestCase.java:85]: Shall we add the other user modes as well and validate. Ex: email usrname

File: modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/base/APIMIntegrationBaseTest.java [hunk 3]
```diff
@@ -70,10 +70,7 @@
 import org.wso2.carbon.tenant.mgt.stub.beans.xsd.TenantInfoBean;
 import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
 
-import java.io.File;
-import java.io.FileNotFoundException;
-import java.io.IOException;
-import java.io.UnsupportedEncodingException;
+import java.io.*;
```
Comment (mushthaq33) [inline modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/base/APIMIntegrationBaseTest.java:73]: Let's remove the * imports

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/soap/SOAPPassThroughTestCase.java [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.net.Socket;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/soap/SOAPPassThroughTestCase.java:51]: * imports

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/soap/SOAPPassThroughTestCase.java [hunk 5]
```diff
@@ -48,7 +47,8 @@
 import java.util.List;
 import java.util.Map;
 
-import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
 import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
 import static org.testng.Assert.*;
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/soap/SOAPPassThroughTestCase.java:53]: There are some more * imports. Let's remove all of it by going through the code

File: modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml [hunk 6]
```diff
@@ -373,4 +373,9 @@
             <class name="org.wso2.am.integration.tests.workflow.WorkflowApprovalExecutorTest"/>
         </classes>
     </test>
+    <test name="apim-integration-tests-soap" preserve-order="true" parallel="false" group-by-instances="true">
+        <classes>
+            <class name="org.wso2.am.integration.tests.other.WSDLImportTestCase"/>
+        </classes>
+    </test>
```
Comment (wasuradananjith) [inline modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml:380]: We do not need to add this here. This is already there under,

```<test name="apim-integration-tests-samples" preserve-order="true" parallel="false" group-by-instances="true">```

Comment (binodmx) [reply modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml]: fixed

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**